### PR TITLE
Add `dataFilter` to filter data on the application side

### PR DESCRIPTION
**Pull request by antho1404**
It is required to be able to execute a task only when specific data are valid on the application side.
This feature is not yet provided by the api but can easily be added on the client side.

### CODE REVIEW

**Comment by ilgooz** [comment]
This implementation of data filters doesn't cover the [Public API Spec of Application](https://github.com/mesg-foundation/core/wiki/Public-API-of-Application-Libraries) but I think, this is something we should do later for all the public APIs of _mesg-js_.

**Comment by antho1404** [comment]
Thanks @ilgooz for the reminder, you're totally right we have to normalize this but I agree this should be another version, now it's just to limit breaking changes. I added some consistency on the api without having any breaking change, can you review it again and after i would like to make a new version of with that

File: src/application/application_test.ts [hunk 1]
```diff
@@ -46,14 +46,14 @@ test('Application should expose gRPC api', (t) => {
 
 const event: Event = {
     serviceID: 'id',
-    filter: '*',
-    dataFilter: (_, eventData) => eventData['foo'] == 'bar'
+    eventKey: '*',
+    filter: (_, eventData) => eventData['foo'] === 'bar'
```
Comment (ilgooz) [inline src/application/application_test.ts]: @antho1404 since we're using a transpiler to convert ts code to ecma5, it already deals with _==_ to _===_

Comment (antho1404) [reply src/application/application_test.ts]: good point, it's just a replace from the merge, done

**Comment by ilgooz** [comment]
lgtm, manual tests works

**Pull request merged**